pymeasure/instruments/teledyne/teledyneHDO6xxx.py

```diff
@@ -0,0 +1,256 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import re
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set
+from pymeasure.instruments.lecroy.lecroyT3DSO1204 import (
+    LeCroyT3DSO1204,
+    ScopeChannel as LeCroyT3DSO1204ScopeChannel,
+)
+
+
+def _channel_results_to_dict(results):
+    """Turn a per-channel result list into a more convenient dict.
+
+    E.g. turn ['C1', 'OFF', 'C2', 'OFF', 'C3', 'OFF', 'C4', 'OFF'] into
+    {'C1': 'OFF', 'C2': 'OFF', 'C3': 'OFF', 'C4': 'OFF'}
+    """
+    keys = results[::2]
+    values = results[1::2]
+    return dict(zip(keys, values))
+
+
+def _remove_unit(value):
+    """Remove a unit from the returned string and cast to float."""
+    if value.endswith(" V"):
+        value = value[:-2]
+
+    return float(value)
+
+
+class ScopeChannel(LeCroyT3DSO1204ScopeChannel):
+    """Extended Channel object for the HDO6xxx."""
+
+    # For some reason "20MHZ" is registered as "ON"
+    _BANDWIDTH_LIMITS = ["OFF", "ON", "200MHZ"]
+
+    _TRIGGER_SLOPES = {"negative": "NEG", "positive": "POS"}
+
+    @property
+    def trigger_level2(self):
```

Comment:
you should be able to delete properties after class definition -- `del ScopeChannel.trigger_level2`.
Although, ideally such properties are pushed down to the class that adds them (i.e. the commonalities are pulled up into a common superclass), as removing class behaviour is considered "bad practice".